### PR TITLE
Add combined function bend element.

### PR DESCRIPTION
**Pull request by cemitch99**
Includes the linear map for a combined-function bend, consisting of a sector bend with a nonzero quadrupole field component.

Related to #374

- [x] Add element particle push.
- [x] Add benchmark example.
- [x] Add Python equivalents.
- [x] Update documentation.
  - [x] link in TOC tree of `examples.rst`

### CODE REVIEW

**Review by qianglbl — APPROVED** [review]

**Comment by ax3l** [comment]
This looks great, thank you!

I fixed the formatting a bit and pushed the changes. Do you like to add a benchmark in this PR or shall we do a follow-up? :)

**Comment by cemitch99** [comment]
> This looks great, thank you!
> 
> I fixed the formatting a bit and pushed the changes. Do you like to add a benchmark in this PR or shall we do a follow-up? :)

I added a benchmark.

File: examples/cfbend/README.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
+.. _examples-cfbend:
+
+Combined Function Bend
+=======================
```
Comment (ax3l) [inline examples/cfbend/README.rst]:
```suggestion
Combined Function Bend
======================
```

File: examples/cfbend/README.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+A single combined function bending magnet (an ideal sector bend with an upright quadrupole field component added).  The magnet
+parameters are based a single CSBEND element appearing in the ELEGANT input file for the ALS-U lattice.
+
+The beam parameters are based on:
+C. Steier et al, "Status of the Conceptual Design of ALS-U", IPAC2017, WEPAB104 (2017).
```
Comment (ax3l) [inline examples/cfbend/README.rst]: I only found this in google scholar:
```suggestion
C. Steier et al, "Status of the Conceptual Design of ALS-U", MEDSI2018, TUPH14, `DOI:10.18429/JACoW-MEDSI2018-TUPH14 <https://doi.org/10.18429/JACoW-MEDSI2018-TUPH14>`__ (2018).
```

File: examples/cfbend/input_cfbend.in [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Particle Beam(s)
+###############################################################################
```
Comment (ax3l) [inline examples/cfbend/input_cfbend.in:3]:
```suggestion
###############################################################################
# Particle Beam(s)
###############################################################################
```

File: examples/cfbend/input_cfbend.in [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+# Beamline: lattice elements and segments
+###############################################################################
+lattice.elements = monitor cfbend1 monitor
+
+lattice.nslice = 25
+
+cfbend1.type = cfbend
+cfbend1.ds = 0.5       # projected length 0.5 m, angle 3.76 deg
+cfbend1.rc = 7.613657587094493   # bending radius [m]
+cfbend1.k = -7.057403   #(upright) quadrupole component [m^(-2)]
```
Comment (ax3l) [inline examples/cfbend/input_cfbend.in]:
```suggestion
cfbend1.k = -7.057403   # (upright) quadrupole component [m^(-2)]
```

File: examples/cfbend/run_cfbend.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022-2023 ImpactX contributors
+# Authors: Marco Garten, Axel Huebl, Chad Mitchell
```
Comment (ax3l) [inline examples/cfbend/run_cfbend.py]: Probably copied?
```suggestion
# Authors: Chad Mitchell, Axel Huebl
```
Feel free to add Marco if he contributed to this example.

File: examples/cfbend/run_cfbend_madx.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022-2023 ImpactX contributors
+# Authors: Marco Garten, Axel Huebl, Chad Mitchell
```
Comment (ax3l) [inline examples/cfbend/run_cfbend_madx.py]: As above
```suggestion
# Authors: Chad Mitchell, Axel Huebl
```

File: examples/cfbend/README.rst (hunk 2 above)
Comment (ax3l) [reply examples/cfbend/README.rst]: Oh, this one? :)
```suggestion
C. Steier et al, "Status of the Conceptual Design of ALS-U", IPAC2017, WEPAB104, `DOI:10.18429/JACoW-IPAC2017-WEPAB104 <https://doi.org/10.18429/JACoW-IPAC2017-WEPAB104>`__ (2017).
```

**Review by ax3l — APPROVED** [review]
Looks great, thank you!

**Pull request merged**